components/fs/fat/fatfs_config.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+// Flag to control whether enabling debug print for fatfs
+// #define FS_DEBUG_PRINT
+
+#define DATA_REGION_SIZE 0x4000000
+
+// Maximum fat partition the file system can have
+#define MAX_FATFS 1
+
+// Maximum opened files
+#define MAX_OPENED_FILENUM 32
+
+// Maximum opened directories
+#define MAX_OPENED_DIRNUM 16
+
+// The number of worker thread, if changes are needed, initialisation of the thread pool must be changes as well
+#define WORKER_THREAD_NUM 4
+
+#define THREAD_NUM (WORKER_THREAD_NUM + 1)
+
+#define WORKER_THREAD_STACKSIZE 0x40000
+
+/*
+ *  This def control whether the memory address passed to the blk device driver should be strict aligned to the 
+ *  BLK_TRANSFER_SIZE and if the BLK_TRANSFER_SIZE should be set to 4KB even if the file system is formatted with
+ *  a different sector size. This setting is for future benchmarking purpose. When using with unmodified SDDF, this should
+ *  always be enabled even though it may cause some performance degradation.
+ *  To enable benchmarking for zero copy, low overhead write, aside from delete the def below, the .system file and blk_config.h
+ *  must be modified to correctly map shared memory between micropython and file system and fs_metadata to blk virt and blk driver.
+ *  SDDF need to be modified as well to remove page alignment check and make sure BLK_TRANSFER_SIZE is equal to the blk device sector
+ *  size.
+ */
+#define MEMBUF_STRICT_ALIGN_TO_BLK_TRANSFER_SIZE
```

Comment:
I'm not sure if we want this to be a configuration option.

Reply:
sDDF sets the policy for block transfer size right now so I don't think the file system should be encouraging people to change it.

Reply:
That is only for comparing the overhead of strict alignment in future benchmarking. I think I have left a comment above that option.

Reply:
I'm not sure if we should be prematurely generalising for benchmarking that hasn't happened yet.

Reply:
Right, I will change it later today, I am trying to get sdmmc driver compiled with sddf_blk and testing it with the file system first, and then I will fix the issues you have mentioned.